docs/index.rst

```diff
@@ -75,20 +99,85 @@ An example of how to read positions from a TNG file is shown below.
          # get the data from the requested block by supplying NumPy array which
          # is updated in-place or returned 
          
-         positons = ts.get_positions(positions)
+         # update in place
+         ts.get_positions(positions)
+         # positions = ts.get_positions(positions) is equivalent
+
+         # or return by value
+         box_vec = ts.get_box(box_vec)
+         # ts.get_box(box_vec) is equivalent
 
            
 
 It is also possible to slice and index the file object to select particular
-frames
+frames individually:
 
 .. code-block:: python
 
   import pytng
-  import numpy as np
+  import numpy as np 
 
   with pytng.TNGFileIterator('traj.tng', 'r') as tng:
-      tng[100].get_positions(positions)
+      positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+      box_vec = tng.make_ndarray_for_block_from_name("TNG_TRAJ_BOX_SHAPE")
+      positions = tng[100].get_positions(positions)
+      box_vec = tng[200].get_box(box_vec)
+
+
+If the step to read is not on the stride of the requested datatype, the NumPy
+array will be returned filled with `np.nan`. A contrived example of this is given below:
+
+.. code-block:: python
+
+   import pytng
+   import numpy as np
+
+   with pytng.TNGFileIterator("traj.tng", 'r') as tng:
+      # make array for positions
+      positions = tng.make_ndarray_for_block_from_name("TNG_TRAJ_POSITIONS")
+      
+      # choose a step
+      step = 42
+
+      # check that we are off stride (stride%step != 0)
+      assert(tng.block_strides["TNG_TRAJ_POSITIONS"]%step != 0 )
+
+      # get the data, which will be returned full of np.nan
+      tng[step].get_positions(positions)
+
+      # check that the read was blank
+      is_blank_read = np.all(np.isnan(positions)) # this will be true
```

Comment:
This is the example that I asked for in an earlier comment (when I hadn't seen this one). 

The question remains how expensive `np.all(np.isnan(positions))` is. Don't the TNG iterator "know" that it filled the array and couldn't it set a flag that could be queried by calling code? Then we wouldn't even have to fill the array, calling code would just have to check the flag.

Reply:
I can add a query-able flag on the integrator step, or alternatively we can possible return a `NamedTuple` with the read status contained as one of the elements.

 I'll play with a few options :)

Reply:
There is now a query-able flag for the success of the last read in `CurrentIntegratorStep.read_success`. Open to other suggestions :)

I think we should still fill the target array with `np.nan` for blank reads, otherwise the current contents of the target array will be the last read step which is potentially misleading when iterating.

I'll do some timings to see how costly filling with `np.nan` is.

Reply:
I'd be curious about the penalty for filling the array. It seems wasteful to do.

Reply:
In MDA we have `has_positions`, `has_velocities`, `has_forces` flags on `Timesteps`. Would it make sense to have a bit array for all content in TNG trajectory that, for each queried step, just has a True or False for content present?

Reply:
Makes sense and could be part of enhancements surrounding iteration. I'll raise an issue.

Reply:
see #47